bin/bootstrap_practice_exercise.exs

```diff
@@ -0,0 +1,282 @@
+
+  def project do
+    [
+      app: :#{exercise_path},
+      version: "0.1.0",
+      # elixir: "~> 1.8",
+      start_permanent: Mix.env() == :prod,
+      deps: deps()
+    ]
+  end
+
+  # Run "mix help compile.app" to learn about applications.
+  def application do
+    [
+      extra_applications: [:logger]
+    ]
+  end
+
+  # Run "mix help deps" to learn about dependencies.
+  defp deps do
+    [
+      # {:dep_from_hexpm, "~> 0.3.0"},
+      # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"}
+    ]
+  end
+end
+"""
+
+Mix.Generator.create_file("exercises/practice/#{exercise}/mix.exs", mix)
+
+# test/test_helper.exs
+test_helper = """
+ExUnit.start()
+ExUnit.configure(exclude: :pending, trace: true)
+"""
+
+Mix.Generator.create_file("exercises/practice/#{exercise}/test/test_helper.exs", test_helper)
+
+## Step 3: write files that depend on problem specifications
+
+url =
+  'https://raw.githubusercontent.com/exercism/problem-specifications/main/exercises/#{exercise}'
+
+:inets.start()
+:ssl.start()
+
+# .docs/instructions.md
+{:ok, {_status, _header, description}} =
+  :httpc.request(:get, {url ++ '/description.md', []}, [], [])
+
+Mix.Generator.create_file("exercises/practice/#{exercise}/.docs/instructions.md", description)
+
+# .meta/config.json
+{:ok, {_status, _header, metadata}} = :httpc.request(:get, {url ++ '/metadata.toml', []}, [], [])
+
+metadata =
+  metadata
+  |> to_string
+  |> Toml.decode!()
+
+config = %{
+  authors: [],
+  contributors: [],
+  files: %{
+    solution: ["lib/#{exercise_path}.ex"],
+    test: ["test/#{exercise_path}_test.exs"],
+    example: [".meta/example.ex"]
+  }
+}
+
+config =
+  Map.merge(metadata, config)
+  |> Jason.encode!(pretty: true)
+
+Mix.Generator.create_file("exercises/practice/#{exercise}/.meta/config.json", config)
+IO.puts("Don't forget to add your name and the names of contributors")
+
+# tests and lib files
+{:ok, {_status, _header, data}} =
+  :httpc.request(:get, {url ++ '/canonical-data.json', []}, [], [])
+
+data = Jason.decode!(data)
+
+properties = Generate.explore_properties(data)
+
+# Generating lib file
+lib_file = """
+defmodule #{module} do
+
+#{properties |> Map.values() |> Enum.map_join("\n\n", &Generate.print_property/1)}
+
+end
+"""
+
+path = "exercises/practice/#{exercise}/lib/#{exercise_path}.ex"
+Mix.Generator.create_file(path, lib_file)
+System.cmd("mix", ["format", path])
+
+Mix.Generator.copy_file(path, "exercises/practice/#{exercise}/.meta/example.ex")
+
+# Generating test file
+test_file =
+  """
+  defmodule #{module}Test do
+    use ExUnit.Case
+
+    #{Generate.print_comments(data, ["cases", "exercise"])}
+    #{Enum.map_join(data["cases"], "\n\n", &Generate.print_test_case(&1, properties, module))}
+  end
+  """
+  |> String.replace("@tag", "# @tag", global: false)
+
+path = "exercises/practice/#{exercise}/test/#{exercise_path}_test.exs"
+Mix.Generator.create_file(path, test_file)
+System.cmd("mix", ["format", path])
```

Comment:
Do you know if there is a way to call `mix format` from within a `.exs` script without having to use `System.cmd`? I searched for it, but couldn't find it.  
Although maybe it's best to run it this way so that it includes the  format in `.formatter.exs`.

Reply:
Every mix task is just a module with a `run` function, so it should be possible to run `Mix.Tasks.Format.run([])`. It should read `.formatter.exs` from the current directory anyway.

mix format documentation: https://hexdocs.pm/mix/master/Mix.Tasks.Format.html
and implementation: https://github.com/elixir-lang/elixir/blob/2a4312412be39303af08ccd0cef81b5bbcd55eac/lib/mix/lib/mix/tasks/format.ex#L1